tests/functional/test_snap_dcgm.py

```diff
@@ -0,0 +1,65 @@
+import json
+import subprocess
+import time
+
+
+def test_dcgm_exporter_endpoint():
+    """Test of the dcgm-exporter service and its endpoint."""
+    endpoint = "localhost:9400/metrics"
+    timeout = 60  # seconds
+    start_time = time.time()
+
+    def query_endpoint():
+        return subprocess.run(["curl", endpoint], text=True, capture_output=True)
+
+    while (result := query_endpoint()).returncode != 0 or not result.stdout.strip():
+        if time.time() - start_time > timeout:
+            assert False, f"Failed to reach '{endpoint}' of the dcgm-exporter service"
+        time.sleep(5)
+
+    assert "DCGM_FI_DRIVER_VERSION" in result.stdout, "No dcgm exported metrics found"
+
+
+def test_dcgm_nv_hostengine():
+    """Check the dcgm-nv-hostengine service."""
+    assert 0 == subprocess.call(
+        ["sudo", "systemctl", "is-active", "--quiet", "snap.dcgm.nv-hostengine.service"]
+    ), "DCGM NV Hostengine service is not running"
+
+
+def test_dcgmi():
+    """Test of the dcgmi command."""
+    result = subprocess.run(["dcgm.dcgmi", "discovery", "-l"], check=True, capture_output=True, text=True)
+    assert "GPU ID" in result.stdout.strip(), "DCGMI is not working"
```

Comment:
Same here, I think this will break to run on CI.

Reply:
I tested it on local VM, it worked for me, just didn't detect any GPUs

Reply:
Yeah....just saw the output and will work:

```
0 GPUs found.
+--------+----------------------------------------------------------------------+
| GPU ID | Device Information                                                   |
+--------+----------------------------------------------------------------------+
+--------+----------------------------------------------------------------------+
0 NvSwitches found.
+-----------+
| Switch ID |
+-----------+
+-----------+
0 CPUs found.
+--------+----------------------------------------------------------------------+
| CPU ID | Device Information                                                   |
+--------+----------------------------------------------------------------------+
+--------+----------------------------------------------------------------------+
```

In any case, I think it would be nice to detect GPU and expect that it's not 0 GPUs found when running in a machine that has a NVDIA and 0 GPUs on a VM for example (could be also checking if there is 0 or more than one GPU ID )